### DISCUSSION TITLE
Interactions in a Sequence

### DISCUSSION THREAD

**Discussion by ravensnowbird** [AaronLasseigne/active_interaction#541] [Ideas]
Hi! I was currently working on a PoC and deciding between `active_interaction` and `interactor` and this is something I would love to see similar in `active_interaction`. Ideally, execute a sequence of Interactions and delegate the errors to the main caller if anything fails in the process.

https://github.com/collectiveidea/interactor
```
class PlaceOrder
  include Interactor::Organizer

  organize CreateOrder, ChargeCard, SendThankYou
end
```

**Comment by ravensnowbird** [comment]
It possible to use `compose` for this but i feel it's much better if we can sequence it in a single line. 
```
class PlaceOrder < ActiveInteraction::Base
  object :params

  def execute
    compose(CreateOrder, params: params)
    compose(ChargeCard, params: params)
    compose(SendThankYou, params: params)
  end
end
```

**Reply by alee-mt** [reply]
With `compose` you do get a response back and can use it. Would this short hand just handle the case where you don't care about the return values?

**Reply by ravensnowbird** [reply]
I was able to use compose to halt the main caller of the interactions.

**Comment by AaronLasseigne** [comment]
Somehow I missed when this was initially posted.

The thing with the single line is you have no way to modify the data between interactors. Some libraries try to get around this by allowing you to create a context that is shared from one interaction to the next. The issue is this can lead to context dependencies that tie interactions together in unfortunate ways. It might be a bit more typing but handling each one gives you full control over how your work with each interaction.
